tests/ObjectPoolTests.cpp

```diff
@@ -0,0 +1,146 @@
+#include "doctestUtils.h"
+
+#include <cesium/omniverse/ObjectPool.h>
+#include <doctest/doctest.h>
+#include <sys/types.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+#include <queue>
+
+#define MAX_TESTED_POOL_SIZE 1024 // the max size pool to randomly generate
+
+// ObjectPool is a virtual class so we cannot directly instantiate it for
+// testing, and instatiating the classes that implement it (FabricGeometryPool
+// and FabricMaterialPool) requires mocking more complicated classes, so we
+// create a bare-bones class here.
+
+class mockObj {
+  public:
+    mockObj(uint64_t objectId) {
+        this->id = objectId;
+        this->active = false;
+    };
+    uint64_t id;
+    bool active;
+};
+
+class mockObjPool final : public cesium::omniverse::ObjectPool<mockObj> {
+  protected:
+    std::shared_ptr<mockObj> createObject(uint64_t objectId) override {
+        return std::make_shared<mockObj>(objectId);
+    };
+    void setActive(const std::shared_ptr<mockObj> obj, bool active) override {
+        obj->active = active;
+    };
+};
+
+void testRandomSequenceOfCmds(mockObjPool* opl, int numEvents, bool setCap) {
+
+    // track the objects we've acquired so we can release them
+    std::queue<std::shared_ptr<mockObj>> activeObjs;
+
+    // the total number of acquires performed, which becomes the minimum
+    // expected size of the pool
+    auto maxActiveCount = opl->getNumberActive();
+
+    // perform a random sequence of acquires/releases while
+    // ensuring we only release what we've acquired
+    for (int i = 0; i < numEvents; i++) {
+        if (!activeObjs.empty() && rand() % 2 == 0) {
+            opl->release(activeObjs.front());
+            activeObjs.pop();
+        } else {
+            activeObjs.push(opl->acquire());
+        }
+        maxActiveCount = std::max(maxActiveCount, activeObjs.size());
+
+        if (setCap && i == numEvents / 2) {
+            // at the halfway point, try resetting the capacity
+
+            // TODO uncomment and place both setCapacity calls in subcases once
+            // assert workaround has been found
+            // see https://github.com/CesiumGS/cesium-omniverse/issues/342
+            // // ensure we don't rollover 0
+            // uint64_t oldCap = opl->getCapacity();
+            // uint64_t guaranteedSmaller = std::min((uint64_t)0, oldCap - 1);
+            // opl->setCapacity(guaranteedSmaller);
+            // // verify that we cannot shrink the capacity
+            // CHECK(oldCap == opl->getCapacity());
```

Comment:
I would just remove this commented code  until #342 is addressed. You can post this snippet as a comment in the issue if you don't want to lose it.